src/porepy/params/data.py

```diff
@@ -1,767 +1,253 @@
+_mathematical_ parameters defined. Solving (systems of) equations with multiple
+instances of the same mathematical parameter (e.g. both thermal diffusivity and
+permeability) is handled by the use of multiple keywords (e.g. "transport" and "flow").
+
+Some default inner dictionaries are provided in pp.params.parameter_dictionaries.
+
+For most instances, a convenient way to set up the parameters is:
+
+    specified_parameters = {pm_1: val_1, ..., pm_n: val_n}
+    data = pp.initialize_data({}, grid, keyword, specified_parameters)
+
+This will assign val_i to the specified parameters pm_i and default parameters to other
+required parameters.
 """
-import warnings
 import numpy as np
-
-from porepy.numerics.mixed_dim.solver import Solver
-from porepy.params.tensor import SecondOrderTensor, FourthOrderTensor
-from porepy.params.bc import BoundaryCondition, BoundaryConditionVectorial
+import porepy as pp
+import numbers
+import warnings
+import porepy.params.parameter_dictionaries as dicts
 
 
-class Parameters(object):
+class Parameters(dict):
     """ Class to store all physical parameters used by solvers.
 
     The intention is to provide a unified way of passing around parameters, and
     also circumvent the issue of using a solver for multiple physical
     processes (e.g. different types of boundary conditions in multi-physics
-    applications).
-
-    List of possible attributes (implemented as properties, see respective
-    getter methods for further description):
-
-    General quantities (not physics specific)
-
-    Scalar quantities
-        biot_alpha: Biot's coefficient in poro-elasticity. Defaults to 1.
-        fluid_viscosity: In a single phase system. Defaults to 1.
-        fluid_compr: Fluid compressibility in a single phase system. Defaults
-            to 0.
-
-    Cell-wise quantities:
-        aperture (fracture width). Defaults to 1.
-        porosity
-
-    Physic-specific
-        tensor (Returns permeability, conductivtiy or stiffness)
-        bc (BoundaryCondition object)
-        bc_val (Boundary values)
-        sources (flow and transport)
-
-    Solvers will access data as needed. If a solver inquires for unassigned
-    data, this will result in a runtime error.
-
-    Attributes (in addition to parameters described above):
-
-    known_physics : list
-        List of keyword signifying physical processes. There are at least one
-        Solver that uses each of the keywords.
+    applications). The keyword assigned to parameters and discretization operators ensures
+    that the right data is used: An operator will always use the parameters stored
+    with under the keyword it has been assigned.
 
+    The parameter class is a thin wrapper around a dictionary. This dictionary contains
+    one sub-dictionary for each keyword.
     """
 
-    def __init__(self, g):
+    def __init__(self, g=None, keywords=None, dictionaries=None):
```

Comment:
The idea was to have it in case we want to add e.g. more sophisticated update methods, which would have to know say the number of faces and the dimension of the grid. But I agree with you, and am not sure if I really want to keep it.